CMakeLists.txt

```diff
@@ -159,86 +158,68 @@ endif()
 ### Ninja build pools
 option(SOFA_NINJA_BUILD_POOLS "Activate the Ninja build pools feature, to limit the cores used by specific targets" OFF)
 
+# Sofa.Config sets the environment (options, compiler flags, global variables)
+sofa_add_module(Sofa/framework/Config Sofa.Config ON)
+
 ### Extlibs
 add_subdirectory(extlibs)
 
-
 ### Compatibility
 option(SOFA_ENABLE_LEGACY_HEADERS "Enable the compatibility layer (headers located in legacy folders) and deprecated cmake modules." ON)
-
 if(SOFA_ENABLE_LEGACY_HEADERS)
     message("Using legacy headers is enabled.")
     message("Headers from v21.12 are usable, but it is advised to change your code to use the new headers introduced in v22.06.")
-
-    add_subdirectory(Sofa/framework/Compat)
-    add_subdirectory(Sofa/Component/Compat)
+    sofa_add_module(Sofa/framework/Compat Sofa.Compat ON)
+    sofa_add_module(Sofa/Component/Compat Sofa.Component.Compat ON)
 endif()
 
 ### SOFA (framework and components)
 add_subdirectory(Sofa)
 
 option(SOFA_WITH_OPENGL "Compile Sofa with OpenGL support. This will define the SOFA_WITH_OPENGL and SOFA_NO_OPENGL CMake variables." ON)
 if(SOFA_WITH_OPENGL)
-    add_subdirectory(${SOFA_KERNEL_SOURCE_DIR}/modules/Sofa.GL)
+    sofa_add_module(${SOFA_KERNEL_SOURCE_DIR}/modules/Sofa.GL Sofa.GL ON)
```

Comment:
Sofa.GL is supposed to be enabled/built before Sofa.GUI could be activated, as some packages inside Sofa.GL needs OpenGL

Reply:
Maybe this is the opportunity to move simply (Sofa.)GL into the Sofa/ folder (similarly to (Sofa,)GUI)